administrator/components/com_associations/models/fields/itemlanguage.php

```diff
@@ -40,33 +40,32 @@ protected function getOptions()
 
 		$input = JFactory::getApplication()->input;
 
-		$referenceId         = $input->get('id', '');
-		$associatedComponent = $input->get('acomponent', '');
-		$associatedView      = $input->get('aview', '');
-		$extension           = $input->get('extension', '');
-		$realView            = $extension !== '' ? $extension : $associatedView;
+		$this->component     = AssociationsHelper::getComponentProperties($input->get('component', '', 'string'));
+		$this->referenceId   = $input->get('id', 0, 'int');
 
-		$key = $extension !== '' ? 'com_categories.category|' . $extension : $associatedComponent . '.' . $associatedView;
-		$cp  = AssociationsHelper::getComponentProperties($key);
-		$associations      = call_user_func(array($cp->associations->gethelper->class, $cp->associations->gethelper->method), $referenceId, $realView);
+		$associations      = call_user_func(
+				array(
+					$this->component->associations->gethelper->class, 
+					$this->component->associations->gethelper->method), 
+					$this->referenceId, $this->component->realcomponent
+			);
 
 		// Get reference language.
-		$table = clone $cp->table;
-		$table->load($referenceId);
-
-		$referenceLanguage = $table->{$cp->fields->language};
+		$table = clone $this->component->table;
+		$table->load($this->referenceId);
 
 		$existingLanguages = JHtml::_('contentlanguage.existing', false, true);
 
 		foreach ($existingLanguages as $key => $lang)
 		{
-			if ($lang->value == $referenceLanguage)
```

Comment:
I was thinking in keeping this for readability, but instead I made a comment :smiley: